liquigraph-core/src/main/java/org/liquigraph/core/io/LockGuardian.java

```diff
@@ -0,0 +1,103 @@
+package org.liquigraph.core.io;
+
+import org.liquigraph.core.exception.LiquigraphLockException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.UUID;
+
+public class LockGuardian {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LockGuardian.class);
+
+    /**
+     * Makes sure the guardian cleans up only its own Lock
+     */
+    private final String uuid;
+
+    public LockGuardian() {
+        uuid = UUID.randomUUID().toString();
```

Comment:
but I can defer toString when I actually pass it to the prepared statement if that's what you mean?